core/templates/dev/head/pages/email_dashboard/EmailDashboardResult.js

```diff
@@ -0,0 +1,91 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for oppia email dashboard page.
+ */
+
+oppia.controller('EmailDashboardResult', [
+  '$scope', '$http', '$window', '$timeout',
+  function($scope, $http, $window, $timeout) {
+    var EMAIL_POST_HANDLER = '/emaildashboardresult';
+
+    $scope.submitEmail = function() {
+      var postLink = EMAIL_POST_HANDLER + '/' + $scope.queryId;
+      var invalidData = false;
+      var EMAIL_DASHBOARD_PAGE = '/emaildashboard';
+
+      if ($scope.email_option !== 'cancel') {
+        if ($scope.email_subject.length === 0) {
+          $scope.invalid.subject = true;
+          invalidData = true;
+        }
+        if ($scope.email_body.length === 0) {
+          $scope.invalid.body = true;
+          invalidData = true;
+        }
+      }
+      if ($scope.email_option === 'custom' &&
+        $scope.num_of_users_to_send === null) {
+        $scope.invalid.num_users = true;
```

Comment:
Try and keep the names consistent -- don't use both `num_users` and `num_of_users_to_send`. (Although I suggested changing this to `max_recipients` -- see above.)